<a href="https://colab.research.google.com/github/iskalinov13/DCVW-Labs/blob/main/LAB8_DCVW.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [15]:
# Assignemnt 8
# Due Date: April, 30th by 10pm

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LogisticRegression
from sklearn import datasets
from sklearn.model_selection import train_test_split
from google.colab import drive
drive.mount("/drive") 

Mounted at /drive


In [20]:
#1. Please donwload IRIS dataset to your notebook. 
#2. Clean data if necessery. 
#3. Split your data into training and testing and apply Logistic Regression. 
#4. Perform cross-validation with k=5 and Logistic Regression
#5. Comment about your resutls from steps 3 and 4. 
PATH = '/drive/My Drive/Colab Notebooks/Iris.csv'
dataset = pd.read_csv(PATH)
X, Y = train_test_split(dataset, 
                       train_size=0.8,
                       test_size=0.2,
                       random_state=42)

In [22]:
X.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
22  23            4.6           3.6            1.0           0.2   
15  16            5.7           4.4            1.5           0.4   
65  66            6.7           3.1            4.4           1.4   
11  12            4.8           3.4            1.6           0.2   
42  43            4.4           3.2            1.3           0.2   

            Species  
22      Iris-setosa  
15      Iris-setosa  
65  Iris-versicolor  
11      Iris-setosa  
42      Iris-setosa

In [24]:
model = LogisticRegression(solver='liblinear', random_state=0)
model.fit(X_train, y_train)

LogisticRegression(random_state=0, solver='liblinear')

In [25]:
predict = model.predict(X_test)
print(predict)

[2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 2 1 0 0 2 2 0 0 2 0 0 1 1 0 2 2 0 2 2 2 0
 2]


In [26]:
model.score(X_test, y_test)

0.868421052631579

In [38]:
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score

kf = KFold(n_splits=5)
scores = cross_val_score(model, X_test, y_test, cv=5, scoring='accuracy')
print(scores)

[0.875      0.875      0.875      0.71428571 0.85714286]


As wee se, in the first 3 iteration the cross validation score is 0.875 .
As we can we, our score is increased from 0.86 -> 0.87 
Cross Validation is used for selecting the best tuning parameters.
